numba/unicode.py

```diff
@@ -509,6 +509,26 @@ def find_impl(a, b):
         return find_impl
 
 
+def unicode_idx_check_type(ty, name):
```

Comment:
this seems to be a duplication of the `unicode_rfind_check_type` below. Can you re-use this function there too and reduce duplication.

Reply:
Reused that in https://github.com/numba/numba/pull/4866.